compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2504,39 +2504,65 @@ genLoadAddressConstant(TR::CodeGenerator * cg, TR::Node * node, uintptr_t value,
       }
 
    TR::Compilation *comp = cg->comp();
-   cursor = generateRILInstruction(cg, comp->target().is64Bit() ? TR::InstOpCode::LLILF : TR::InstOpCode::IILF, node, targetRegister, static_cast<uint32_t>(value), cursor);
+   TR_ResolvedMethod * method = comp->getCurrentMethod();
 
-   bool assumePatch = false;
-   bool isCompressedClassPointer = false;
-   if (node->isClassUnloadingConst())
+   if (node->getOpCodeValue() == TR::aconst &&
+       ((node->isClassPointerConstant() && cg->fe()->isUnloadAssumptionRequired(reinterpret_cast<TR_OpaqueClassBlock *>(value), method)) ||
```

Comment:
Can you explain the motivation behind this check? I do understand that if the node is ClassPointerConstant and would need UnloadAssumption is needed, then instruction should be generated which can be patched. Just question, code seems to be OK.

Reply:
I've refactored this since lots of the checks here are repeated across `genLoadAddressConstant`